cardano-crypto-class/src/Cardano/Crypto/PinnedSizedBytes.hs

```diff
@@ -385,9 +385,12 @@ ptrPsbToSizedPtr = SizedPtr . castPtr
 --   then this throws an exception.
 pinnedByteArrayFromListN :: forall a. Prim.Prim a => Int -> [a] -> ByteArray
 pinnedByteArrayFromListN 0 _ =
-    die "pinnedByteArrayFromListN" "list length zero"
+    die "pinnedByteArrayFromListN" "list length zero #1"
 pinnedByteArrayFromListN n ys = runST $ do
-    marr <- newPinnedByteArray (n * Prim.sizeOf (head ys))
+    let headYs = case ys of
+                  [] -> die "pinnedByteArrayFromListN" "list length zero #2"
+                  (y:_) -> y
+    marr <- newPinnedByteArray (n * Prim.sizeOf headYs)
```

Comment:
With `ghc-9.8` the use of `head` triggers a partial function warning.

Reply:
Not a fan of those `#1` / `#2` markers - they may be useful to track the error back to the source code, but they don't make sense unless you actually look at the source code, which is an unnecessary burden for a user of the library. I'd change them to something more descriptive, e.g.: `"specified size is zero"` when the first argument is `0`, and `"list length zero less than specified size"` when the first argument is nonzero but the list is empty - this gives the caller an immediate clue as to what they need to fix to make this function work (specify a nonzero list length, vs. making sure specified list length and actual list length match).

You can also avoid defining `headYs` by adding another top-level pattern matching, like so:
```haskell
pinnedByteArrayFromListN _ [] =
    die "pinnedByteArrayFromListN" "list length zero less than specified size"
pinnedByteArrayFromListN n ys@(y:_) =
    marr <- newPinnedByteArray (n * Prim.sizeOf y)
    ...
```

Then again, the only use site of this function is inside `psbFromBytes`, which itself is deprecated and not used anywhere in the library, so this might not actually be important enough to fix "properly".

Reply:
Your suggestions results in an "Pattern match(es) are non-exhaustive" error.

Currently using at:
```haskell
--   then this throws an exception.
pinnedByteArrayFromListN :: forall a. Prim.Prim a => Int -> [a] -> ByteArray
pinnedByteArrayFromListN 0 _ =
    die "pinnedByteArrayFromListN" "list length zero"
pinnedByteArrayFromListN _ [] =
    die "pinnedByteArrayFromListN" "list length zero"
pinnedByteArrayFromListN n ys@(y:_) = runST $ do
    marr <- newPinnedByteArray (n * Prim.sizeOf y)
```